Updating monitoring.md with Grafana Config Info (#1589)
CDCgov/phdi: containers/orchestration/monitoring.md

* _Logs_: Logs are the most self-explanatory kind of telemetry data, the usual fare of print and debug statements a service emits to record that an event has happened.

### OpenTelemetry Basics and Configuration

* _Instrumentation_: Instrumentation refers to the process of setting up a service to emit telemetry data (note that instrumentation refers only to _producing_ telemetry data, not doing anything with it). Instrumentation can be either manual or automatic (there's also a third option, programmatic, but for our purposes we won't be using that at all), or some combination of both.
* _Automatic Instrumentation_: This type of instrumentation allows a bootstrapped distribution of OpenTelemetry to scan a service's codebase and, on the back end, add in all the infrastructure context for producing telemetry data. Telemetry producers and consumers are created, combined, and wired together with a global context to make each API endpoint of a service emit a suite of standard information (including traces for all incoming requests, spans for request attributes, and metrics for application health and system performance).

> * _Manual Instrumentation_: This type of instrumentataion must be configured and implemented by a developer. In any case where you'd like to collect and emit information that isn't automatically collected--or you'd like to customize how a particular type of information is emitted--manual instrumentation is required. To instrument a piece of an application, you need three things: a Provider, an Agent, and an Instrument. These things aren't hard to set up, but do require invoking in the correct order.

Reviewer: Just caught a typo of my own making that we might as well correct here: there's an extra 'a' at the end of instrumentation

## How We've Set Things Up

* We start by using OpenTelemetry's automatic instrumentation to create all of the Provider-level instrumentation contexts for the Orchestration Service. This is executed in the project's `Dockerfile`.
* We supplement the automatic instrumentation with manual instrumentation around specific status code formatting for message processing endpoints. This is instrumented in `main.py`.
* We spin up local instances of the OTel Collector, Prometheus, and Grafana, making sure to expose specific relative ports and endpoints. This is executed in the project's `docker-compose.yml`.
* We configure the OTel Collector with the listening/transmitting mechanisms we want to use, and then activate the Collector with a service pipeline. This is set up in `otel-collector-config.yaml`.
* We configure Prometheus with a scrape job to fetch the metrics that the OTel Collector put together for us based on our instrumentation. This is configured in `prometheus.yml`.
* We link Prometheus to Grafana as its database source so that it can pull metrics from Prometheus' local DB into visualization dashboards.

> * We configure `grafana/datasources/datasources.yml` to specify the Prometheus connection as the default DB to use for the visualization in Grafana.

Reviewer 2: ```suggestion
* We configure `grafana/datasources/datasources.yml` to specify Prometheus connection as the default data source to use for the visualization in Grafana.
```

just to match the language further down in the document

> * We configure `grafana/dashboards/dashboard.yml` for the Dashboards we use throughout Grafana. New dashboards can be created in Grafana UI then downloaded as JSON and saved within the `grafana/dashboard` folder for use by any user logging into `localhost:4000`.

Reviewer 2: ```suggestion
* We configure `grafana/dashboards/dashboard.yml` for the Dashboards we use throughout Grafana. New dashboards can be created in Grafana UI then downloaded as JSON and saved within the `grafana/dashboards` folder for use by any user logging into `localhost:4000`.
```

### `grafana.ini`

The primary configuration file for Grafana specifices various settings to customize and secure Grafana. 

* `paths` defines the directory paths for Grafana components, such as where it stores data, logs, plugins, and provisioning.
* `server` defines settings related to the Grafana webserver, including the port, domain name, and public URL.
* `database` is the configuration for Grafana's own database, which is separate from Prometheus' database.
* `log` controls the logging mechanism. Right now, logs are output to console and at the info level.
* `security` settings is where the admin_user and admin_password can be specified.
* `users` are user-specific settings. Right now, we allow users to sign-in and create accounts, but we may want to revisit if we want to create a user hierarchy (admin, read-only user, etc.).

#### `grafana/dashboards/dashboard.yml`

> This is a simple yaml file that specifies that dashboards saved here should be pulled in and loaded to our Grafana UI. New dashboards can be created in the UI then downloaded and saved here as JSON.

Reviewer 2: ```suggestion
This is a simple yaml file that specifies that dashboards saved here should be pulled in and loaded to the Grafana UI. New dashboards can be created in the UI then downloaded and saved here as JSON.
```
